source/Track.cpp

```diff
@@ -0,0 +1,103 @@
+/* Track.cpp
+Copyright (c) 2022 by RisingLeaf
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Track.h"
+
+#include "DataNode.h"
+
+#include <iostream>
+
+using namespace std;
+
+Track::Track(const DataNode &node)
+{
+	Load(node);
+}
+
+
+
+void Track::Load(const DataNode &node)
+{
+	// All tracks need a name.
+	if(node.Size() < 2)
+	{
+		node.PrintTrace("Error: No name specified for track:");
+		return;
+	}
+
+	if(!name.empty())
+	{
+		node.PrintTrace("Error: Duplicate definition of track:");
+		return;
+	}
+	name = node.Token(1);
+
+	for(const DataNode &child : node)
+	{
+		if(child.Token(0) == "volume" && child.Size() >= 2)
+			volumeModifier = child.Value(1);
+		else if(child.Token(0) == "idle" && child.Size() >= 2)
+			idleTitle = child.Token(1);
+		else if(child.Token(0) == "combat" && child.Size() >= 2)
+			combatTitle = child.Token(1);
+		else if(child.Token(0) == "landed" && child.Size() >= 2)
+			landedTitle = child.Token(1);
+		else if(child.Token(0) == "wait" && child.Size() >= 2)
+			wait = max<int>(0, child.Value(1));
+	}
+	finishedLoading = true;
+}
+
+
+
+const std::string Track::Name() const
+{
+	return name;
+}
+
+
+
+const std::string Track::GetTitle(GameState state) const
+{
+	switch(state)
+	{
+		case GameState::IDLE:
+			return idleTitle;
+			break;
+		case GameState::COMBAT:
+			return combatTitle;
+			break;
+		case GameState::LANDED:
+			return landedTitle;
+			break;
+		default:
+			return idleTitle;
+			break;
```

Comment:
mb make the default one also the IDLE one by moving it lower?
does that actually matter performance wise? I'm not sure but I think the switch only looks at it once anyway?

Reply:
This could be an enhancement; I would like to avoid changing the original code as much as possible beyond getting it into a workable state.

If you do experience a performance issue in testing it should be addressed before merging

Reply:
I think you misunderstood, I simply meant we could merge default and IDLE
and I was asking if it wouldnt be a bad idea perf wise

Reply:
Considering GameState only has three states I don't see how `default` would ever get triggered; does it make sense to just delete it entirely?

Reply:
I mean yeah sure

Reply:
Worth noting I didn't write most of the code; just adopted it and am cleaning it up.  Is there a scenario where it is ever possible for one of the three game states to not be one of the three game states passed as an option?

Reply:
i also just noticed that we dont need the break; in any line